### PR TITLE
feat(iterable): user deletion support

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change

- Iterable user deletion support

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by koladilip** [comment]
@mihir-4116 better put destination name in pr title

**Review by koladilip — DISMISSED** [review]

File: src/v0/destinations/iterable/deleteUsers.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        },
+      };
+      const resp = await httpDELETE(url, requestOptions, {
+        destType: 'iterable',
+        feature: 'deleteUsers',
+      });
+      const handledDelResponse = processAxiosResponse(resp);
+      if (!isHttpStatusSuccess(handledDelResponse.status) && handledDelResponse.status !== 404) {
+        if (handledDelResponse.status !== 400) {
+          throw new NetworkError(
```
Comment (koladilip) [inline src/v0/destinations/iterable/deleteUsers.js:43]: why do we need throw this?

Comment (mihir-4116) [reply src/v0/destinations/iterable/deleteUsers.js:43]: these are generic errors which will fail for all users, like invalid api key

Comment (koladilip) [reply src/v0/destinations/iterable/deleteUsers.js:43]: can you add comment for better clarity

**Review by koladilip — DISMISSED** [review]

**Review by koladilip — APPROVED** [review]

**Review by aashishmalik — APPROVED** [review]

**Pull request merged**